### PR TITLE
*: support mock campaign leader in owner manager

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: ref #41495

Problem Summary:

At present, it is not easy to mock the distributed environment in unit tests. The correctness of this feature depends heavily on the integration test in real clusters. As a result, a bugfix with little changes require a whole real cluster to verify, which is quite inefficient.

### What is changed and how it works?

This PR is the first step to mock distributed environment in unit tests. First of all, we mock the leader campaign in the owner manager used for tests.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: domain/domain.go [hunk 1]
```diff
@@ -2153,6 +2153,7 @@ func (do *Domain) UpdateTableStatsLoop(ctx, initStatsCtx sessionctx.Context) err
 		do.wg.Run(do.loadStatsWorker, "loadStatsWorker")
 	}
 	owner := do.newOwnerManager(handle.StatsPrompt, handle.StatsOwnerKey)
+	do.wg.Run(func() { quitStatsOwner(do, owner) }, "quitStatsOwner")
```
Comment (zimulala) [inline domain/domain.go:2156]: Why do it here?  It seems that we do it in `updateStatsWorkerExitPreprocessing`(in `updateStatsWorker`).

Comment (tangenta) [reply domain/domain.go:2156]: If `domain.statsLease` is zero, `updateStatsWorker` is not invoked.

Comment (zimulala) [reply domain/domain.go:2156]: Do we need to consider unified handling it, such as the ` updateStatsWorkerExitPreprocessing ` delete inside.

**Review by zimulala — APPROVED** [review]
LGTM

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**